src/fees/FeeCalculatorBase.sol

```diff
@@ -3,24 +3,37 @@ pragma solidity ^0.8.0;
 
 import {IRouter} from '../interfaces/IRouter.sol';
 
+/// @title Fee calculator base
+/// @notice An abstract contract that provides basic functionality for calculating fees
 abstract contract FeeCalculatorBase {
     error InvalidSender();
     error InvalidRate();
 
+    /// @dev Denominator for calculating basis points
     uint256 internal constant _BPS_BASE = 10_000;
+
+    /// @notice Immutable address for recording the router address
     address public immutable router;
 
-    uint256 public feeRate; // In bps, 20 means 0.2%
+    /// @notice Fee rate in basis points with 20 representing 0.2%
+    uint256 public feeRate;
 
+    /// @dev Initialize the router address and fee rate
     constructor(address router_, uint256 feeRate_) {
         router = router_;
         feeRate = feeRate_;
     }
 
+    /// @notice Calculate the fee for the given amount
+    /// @param amount The amount for which the fee needs to be calculated
+    /// @return The calculated fee
```

Comment:
Because the param `amount` here is fee included, I think we should mention this. What do you think ?

```suggestion
    /// @param amount The amount which included the fee already
    /// @return The fee calculated from `amount`
```

Reply:
Fixed. I think that 'from amount' is not strictly necessary, feel free to comment if you strongly prefer it.

Reply:
looks good.